tests/system/apps/__init__.py

```diff
@@ -96,3 +96,36 @@ def ipv6_healthcheck(app_id=None, parent_group="/"):
         For more information about the nc options just run `docker run alpine nc --help`
     """
     return load_app('ipv6-healthcheck', app_id, parent_group)
+
+
+def faultdomain_app(region=None, zone=None, instances=1, constraints=[], suffix=None):
+    """
+    This is a dynamic app definition based on the faultdomain-base-app. It modifies it by appending
+    the name, zone and region configuration as given
+
+    :param region: The region to start this app in
+    :param zone: The zone to start this app in
+    :param instances: The number of instances in this app
+    :param constraints: Other constraints to append
+    :return: Returns the App Definition
+    """
+    app = load_app('faultdomain-base-app',
+                   app_suffix=("" if suffix is None else "-{}".format(suffix)))
```

Comment:
I'm not sure we need the suffix.

Reply:
Because depending on the `region`, `zone` and `constraints` you can create different variations of the app definition. The `suffix` is a symbolic name for the region/zone/constraints combination so it's easier debuggable.